src/ucp/core/ucp_ep.c

```diff
@@ -163,9 +163,11 @@ void ucp_ep_config_key_reset(ucp_ep_config_key_t *key)
     memset(key->amo_lanes,    UCP_NULL_LANE, sizeof(key->amo_lanes));
 }
 
-ucs_status_t ucp_ep_create_base(ucp_worker_h worker, const char *peer_name,
-                                const char *message, ucp_ep_h *ep_p)
+ucs_status_t ucp_ep_create_base(ucp_worker_h worker, unsigned ep_init_flags,
```

Comment:
The method has become too large. I'd like to suggest to move some parts of code to separate methods. E.g. ep_init, ep_ext_init, ep_insert_to_worker, and complex conditions can be replaced by call to method, e.g. ep_has_indirect_id.

Reply:
done